### PR TITLE
Fix timepicker update

### PR DESCRIPTION
**Pull request by daytonn**
This solves a source of truth error in the date picker modal for promotions. The tricky part was that we need to set the date-time with two separate components, one that takes a `Date` object and one that takes a `moment` date.

This solves the issue by creating a third value of `dateTime` whose value is the combination of both date and time. This way the interface can deal with `date` and `time` in isolation, and behind the scenes the `dateTime` can be updated by these changes.

This is the reason that 2-way binding is a bad idea, thanks angular/ember

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,222 @@
+{
```
Comment (kand) [inline .eslintrc]: Does the project currently pass all these lint rules? I'd rather not change the way the project is linted in this PR.

If we do want to change the way the project is linted, we should do that in a separate PR.

**Comment by daytonn** [comment]
[WEBTECH - promotions modal](https://trello.com/c/eFqRKNjN/641-webtech-promotions-modal)

File: app/components/date-time-format-filter/date-time-format-filter.js [hunk 2]
```diff
@@ -3,22 +3,30 @@
...
   'lodash',
-  'moment'
+  'moment',
 ])
   .filter('dateTimeFormat', [
     '_', 'moment', 'CmsConfig',
     function (_, moment, CmsConfig) {
-      return function (date, format) {
+      function isInvalidDateValue (dateValue) {
+        return !(
```
Comment (kand) [inline app/components/date-time-format-filter/date-time-format-filter.js]: Too much analysis is required to understand this statement, distribute the not operator so this is more readable.

```
return !_.isString(dateValue) && !_.isDate(dateValue) && !moment.isMoment(dateValue);
```

Or make this a positive function `isValidDateValue` and do `!isValidDateValue` when doing the check.

Also is `isInvalidDateValue('my string value') === false`? If so, this check should be more specific about what a valid `String` is.

Checking the moment docs, you might be able to just use [`moment(<date-like-thing>).isValid()`](http://momentjs.com/docs/#/parsing/is-valid/) or [`moment.invalid(<date-like-thing>)`](http://momentjs.com/docs/#/utilities/invalid/).

File: app/components/date-time-modal/date-time-modal-controller.js [hunk 3]
```diff
@@ -3,85 +3,81 @@
...
-          $scope.tempDatetime.minute (newTime.minute());
-          $scope.tempDatetime.second(newTime.second());
-        }
+      $scope.$watch('date', function () {
+        var date = $scope.date.clone();
+        var time = moment($scope.time);
+        var newDate = moment()
+          .dayOfYear(date.dayOfYear())
+          .hours(time.hours())
+          .minutes(time.minutes());
```
Comment (kand) [inline app/components/date-time-modal/date-time-modal-controller.js]: Do we need to set `year` here and in the `time` watch as well?

File: app/components/date-time-modal/date-time-modal-controller.tests.js [hunk 4]
```diff
@@ -32,194 +35,189 @@ describe('Controller: DatetimeSelectionModalCtrl', function () {
...
   afterEach(function () {
     sandbox.restore();
   });
 
-  it('should set a timezone label on scope', function () {
+  it('has a TIMEZONE_LABEL', function () {
     CmsConfigProviderHook.setTimezoneName('America/Chicago');
-
     buildControllerInstance();
-
```
Comment (kand) [inline app/components/date-time-modal/date-time-modal-controller.tests.js]: I like to have spaces between setup, execution, and assertion for readability. Arguable, but I'd prefer to keep the spaced style to match newer tests.

File: karma.conf.js [hunk 5]
```diff
@@ -16,6 +16,13 @@ module.exports = function(config) {
   config.set({
     // base path, that will be used to resolve files and exclude
     basePath: '',
+    plugins: [
+      'karma-coverage',
+      'karma-mocha',
+      'karma-ng-html2js-preprocessor',
+      'karma-sinon-chai',
+      'karma-chrome-launcher',
```
Comment (kand) [inline karma.conf.js]: What is this addition a response to?

File: .eslintrc (hunk 1 above)
Comment (daytonn) [inline .eslintrc]: I'm not changing how the project is linted, I just added an .eslintrc file because my editor is configured with eslint.

File: app/components/date-time-format-filter/date-time-format-filter.js (hunk 2 above)
Comment (daytonn) [inline app/components/date-time-format-filter/date-time-format-filter.js]: I changed it to several negations. I just needed this function to be tolerant of `Date` values. Previously it assumed the passed in value was a moment object or a date string. The changes allow you to pass a valid date, date string, or moment object and get the same format behaviour.

File: karma.conf.js (hunk 5 above)
Comment (daytonn) [inline karma.conf.js]: I wanted to run karma from the command line without grunt making it hard to work with. I'm not sure why this isn't needed with the grunt setup but you need to declare these plugins to work with karma directly.

File: app/components/date-time-modal/date-time-modal-controller.js (hunk 3 above)
Comment (daytonn) [inline app/components/date-time-modal/date-time-modal-controller.js]: I added the year in.

File: .eslintrc (hunk 1 above)
Comment (daytonn) [inline .eslintrc]: I could ignore it but that could be confusing if we move to esint in the future.

File: karma.conf.js (hunk 5 above)
Comment (daytonn) [inline karma.conf.js]: Worth noting is that using Grunt to run karma for some reason removes the ability to run the tests in the browser, which is a deal-breaker IMO.

File: .eslintrc (hunk 1 above)
Comment (kand) [inline .eslintrc]: Alright, as long as `./scripts/lint` passes.
